### PR TITLE
🐛 Fix LET Colorization

### LINKED ISSUE #1198: 🐛 LET: Boundary Event and Gateway Colorization Is Wrong
#### Describe your issue

**Have a look at https://github.com/process-engine/process_engine_runtime/issues/187 as well.**

When using boundary events in some cases they get colored even though they should not.

This is because the boundary events do not have a token history, instead the let kind of guesses if the boundary event was used or not.

**To be exact:**
If the element after the boundary event has a token and the task on which the boundary event sticks has no token history or is still active the boundary event will be marked.

#### How can others reproduce the issue?

- Run [this](https://github.com/process-engine/bpmn-studio/files/2676438/LET-test-Boundary.bpmn.zip) diagram
- **Notice that the Boundary Event will be marked even though it should not.**

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by SteffenKn**
**Changes:**

1. Fix Colorization of BoundaryEvents and Gateways
2. Fix Colorizing Outgoing SequenceFlows of Suspended Tasks
3. Fix Colorization for Multiple Outgoing Elements
4. Fix Colorization of SequenceFlows When using Loops
5. Fix Colorization of SequenceFlow With ParallelGateway as Target

**Issues:**

Closes #1198

PR: #1332

## How can others test the changes?

- Open up the BPMN-Studio 
- Run a diagram with BoundaryEvents, Gateways, ParallelGateways, Tasks and Loops
- **Notice that everything should be colored as expected.**

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,8 +108,8 @@
     "@process-engine/consumer_api_contracts": "4.1.1",
     "@process-engine/kpi_api_contracts": "1.2.3",
     "@process-engine/management_api_client": "3.2.2",
-    "@process-engine/management_api_contracts": "4.2.1",
-    "@process-engine/process_engine_runtime": "6.0.0",
+    "@process-engine/management_api_contracts": "4.2.0",
+    "@process-engine/process_engine_runtime": "6.0.0-275644be-b1",
```
Comment (Vyperus) [inline package.json]:
```suggestion
    "@process-engine/process_engine_runtime": "6.0.0",
```

**Comment by ElRaptorus** [comment]
@SteffenKn What @Vyperus said. Looks good to me otherwise.

File: src/modules/live-execution-tracker/live-execution-tracker.ts [hunk 2]
```diff
@@ -533,68 +535,74 @@ export class LiveExecutionTracker {
...
-      } else {
-        elementWithIncomingElements.push(incomingElementAsShape);
+        // This is needed because the ParallelGateway only knows the flowNodeId of the first element that reaches the ParallelGateway
+        const targetOfOutgoingElementIsGateway: boolean = targetOfOutgoingElement.type === 'bpmn:ParallelGateway';
 
-        const sourceHasNoTokenHistory: boolean = !this._hasElementTokenHistory(sourceOfIncomingElement.id, tokenHistoryGroups);
-        if (sourceHasNoTokenHistory) {
-          elementWithIncomingElements.push(sourceOfIncomingElement);
+        const sequenceFlowWasExecuted: boolean = previousFlowNodeInstanceIdOfTarget === flowNodeInstanceIdOfElement;
+        if (sequenceFlowWasExecuted || targetOfOutgoingElementIsGateway) {
```
Comment (cmg-dev) [inline src/modules/live-execution-tracker/live-execution-tracker.ts]: Theres is another business rule. Extract this please.

```suggestion
       const needToAddToOutgoingElements: boolean  = sequenceFlowWasExecuted || targetOfOutgoingElementIsGateway
        if (needToAddToOutgoingElements) {
```

File: package.json [hunk 3]
```diff
@@ -108,7 +108,7 @@
     "@process-engine/consumer_api_contracts": "4.1.1",
     "@process-engine/kpi_api_contracts": "1.2.3",
     "@process-engine/management_api_client": "3.2.2",
-    "@process-engine/management_api_contracts": "4.2.1",
+    "@process-engine/management_api_contracts": "4.2.0",
```
Comment (ElRaptorus) [inline package.json]: Is this intentional or a rebase error?

Comment (SteffenKn) [reply package.json]: Looks like a rebase error 👍

**Pull request merged**